docs/developer-guide/ksqldb-clients/java-client.md

```diff
@@ -767,6 +768,35 @@ System.out.println(description.name()
 );
 ```
 
+Use variable substitution<a name="variable-substitution"></a>
+---------------------------------------------------------------
+
+Starting with ksqlDB 0.18.0, users can define session variables by calling the `define()` method and
+reference them in other fuctions by wrapping the variable name in `${}`. The `undefine()` method
+undefines a session variable, and `getVariables()` returns a map of the currently defined variables
+and their values.
+
```

Comment:
These docs need to specify which client methods support variable substitution. Otherwise, users might be confused that variable substitution isn't working for, e.g., the `describeSource()` method. Let's sanity-check the desired behavior here with product as well. I can see a strong case for supporting substitution for all methods, including `describeSource()`.